rdr_service/dao/study_nph_dao.py

```diff
@@ -506,54 +487,77 @@ def _is_order_cancelled(order: Order) -> bool:
+                "specimenCode": (ordered_sample.identifier or ordered_sample.test),
+                "timepointID": nph_study_category_dao.get_study_category(order.category_id).name,
+                "volume": ordered_sample.volume,
+                "volumeUOM": ordered_sample.volumeUnits,
+                "orderedSampleStatus": sample_status,
+                "clientID": order.client_id,
+                "collectionDateUTC": collection_date_utc,
+                "processingDateUTC": processing_date_utc,
+                "finalizedDateUTC": finalized_date_utc,
+                "sampleID": (ordered_sample.aliquot_id or ordered_sample.nph_sample_id),
+                "kitID": kit_id,
+                "biobankStatus": None,
+            }
+            biobank_status_and_lims_id = (
+                nph_stored_sample_session.get_biobank_status_and_lims_id(
+                    nph_participant, ordered_sample
                 )
-                kit_id = None
-                if (ordered_sample.identifier or ordered_sample.test).startswith("ST"):
-                    kit_id = order.nph_order_id
-
-                sample_status = "Cancelled" if sample_is_cancelled else "Active"
-                biospecimen_dict = {
-                    "orderID": order.nph_order_id,
-                    "visitID": parent_study_category.name if parent_study_category else "",
-                    "studyID": f"NPH Module {nph_module_id.name}",
-                    "specimenCode": (ordered_sample.identifier or ordered_sample.test),
-                    "timepointID": nph_study_category_dao.get_study_category(order.category_id).name,
-                    "volume": ordered_sample.volume,
-                    "volumeUOM": ordered_sample.volumeUnits,
-                    "orderedSampleStatus": sample_status,
-                    "clientID": order.client_id,
-                    "collectionDateUTC": collection_date_utc,
-                    "processingDateUTC": processing_date_utc,
-                    "finalizedDateUTC": finalized_date_utc,
-                    "sampleID": (ordered_sample.aliquot_id or ordered_sample.nph_sample_id),
-                    "kitID": kit_id,
-                    "biobankStatus": None,
-                }
-                biobank_status_and_lims_id = (
-                    nph_stored_sample_session.get_biobank_status_and_lims_id(
-                        nph_participant, ordered_sample
+            )
+            if biobank_status_and_lims_id:
+                biospecimen_dict.update({"biobankStatus": biobank_status_and_lims_id})
+            yield biospecimen_dict
+
+    def get_nph_biospecimens_for_participant(self, nph_participant: Participant):
+        with self.session() as session:
+            participant: Participant = session.query(Participant).filter(
+                    Order.participant_id == nph_participant.id
+                )\
+                .options(joinedload(Participant.orders).joinedload(Order.samples)).first()
```

Comment:
suggesting change for readability

```suggestion
            participant: Participant = session.query(Participant).filter(
                Order.participant_id == nph_participant.id
            ).options(
                joinedload(Participant.orders).joinedload(Order.samples)
            ).first()
```